app/code/Transoft/Blog/view/frontend/templates/blog_post_list.phtml

```diff
@@ -0,0 +1,34 @@
+<?php
+use Transoft\Blog\Block\Bloglist;
+
+?>
+<?php /** @var Bloglist $block*/
+?>
+<?php
+const DATA_SEPARATOR = '_sep_';
+
+$bloglist = $block->getData('bloglist');
+$data = $bloglist->getLatestPostItems();
+?>
+
+<?php if ($bloglist->isTypesMatching()): ?>
+    <div class="blog-list">
+    </div>
+    <script type="text/x-magento-init">
+    {
+        ".blog-list": {
+            "blogListWidget": {
+            <?php foreach ($data as $value): ?>
+                <?php $id = $value->getBlogId();
+                $widgetBlogData = $block->escapeHtml($value->getTheme()) . DATA_SEPARATOR
+                . $block->escapeUrl($bloglist->getUrlById($id));
+                ?>
+                "blog<?= $id ?>": "<?= $widgetBlogData ?>",
+            <?php endforeach; ?>
+                "title": "<?= $block->escapeHtml(__('Latest 5 blogs:')) ?>",
+                "dataSeparator": "<?= DATA_SEPARATOR ?>"
+            }
+        }
```

Comment:
Please return widget data from block. Create method getConfig and return json encoded data

Reply:
done